src/Modal/Modal.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from "react";
+import "./modal-style.scss";
+
+interface ModalProps {
+    toggle: boolean;
+    id?: string;
+    fullscreen?: boolean;
+    position?: "left" | "right";
+    className?: string;
+    disableBackdropDismiss?: boolean;
+    ariaLabel?: string;
+    ariaDescribedby?: string;
+    dismissModal?: () => void;
+    header?: React.ReactNode;
+    body?: React.ReactNode;
+    footer?: React.ReactNode;
+}
+
+interface ModalState {
+    open: boolean;
+    close: boolean;
+}
+
+export class Modal extends React.Component<ModalProps, ModalState> {
+    constructor(props: ModalProps) {
+        super(props);
+        this.state = {
+            open: false,
+            close: false
+        };
+    }
+
+    componentDidMount(): void {
+        if (this.props.toggle === true) {
+            this.setState({ open: true });
+        }
+    }
+
+    componentDidUpdate(prevProps: ModalProps): void {
+        if (this.props.toggle !== prevProps.toggle) {
+            this.setState({ open: !prevProps.toggle, close: prevProps.toggle });
+        }
+    }
+
+    /**
+     * set modal class names
+     * @returns string of concatenated class names
+     */
+    setModalClasses(): string {
+        const classes: string = this.classNames({
+            "show": this.state.open,
+            "fade": this.state.close,
+            "modal-aside": !!this.props.position,
+            "modal-aside-left": this.props.position === "left",
+            "modal-aside-right": this.props.position === "right",
+            "modal-fullscreen": this.props.fullscreen,
+            [this.props.className]: !!this.props.className
+        });
+        return classes;
+    }
+
+    /**
+     * Convert object to string
+     * @param { [key: string]: boolean } classes list of classes to be appended
+     * @returns {string} string of concatenated class names
+     */
+    classNames(classes: { [key: string]: boolean }): string {
+        return Object.entries(classes)
+            .filter(([key, value]) => value)
+            .map(([key, value]) => key)
+            .join(" ");
+    }
+
+    /**
+     * emit dismissModal event when backdrop is clicked
+     * @param {React.MouseEvent} event clicked element
+     */
+    closeModal(event: React.MouseEvent): void {
+        const target: HTMLElement = event.target as HTMLElement;
```

Comment:
you don't need this, just declare `event: React.MouseEvent<HTMLDivElement>` and you should have access to target.

Reply:
`const target: HTMLElement = event.target as HTMLElement ` will solve the issue with line 80 `target.classList`

Reply:
I'm trying to avoid casting the `event.target as HTMLElement` since we don't know why update in the future might we get that changes that type since these events are **[React synthetic events](https://reactjs.org/docs/events.html)**. In order to access the properties of the target, you use this `event.currentTarget.classList` which will inherit the properties of the element you pass to the type constructor which in this case `React.MouseEvent<HTMLDivElement>` takes the properties of the `Div` element.